src/pykeen/triples/instances.py

```diff
@@ -112,31 +148,30 @@ def __init__(
     def __len__(self) -> int:  # noqa: D105
         return self.mapped_triples.shape[0]
 
-    def __getitem__(self, item: int) -> SLCWASampleType:  # noqa: D105
+    def __getitem__(self, item: int) -> SLCWABatch:  # noqa: D105
         positive = self.mapped_triples[item].unsqueeze(dim=0)
         # TODO: some negative samplers require batches
         negative, mask = self.sampler.sample(positive_batch=positive)
         # shape: (1, 3), (1, k, 3), (1, k, 3)?
-        return positive, negative, mask
+        # TODO: weights
+        return SLCWABatch(positives=positive, negatives=negative, masks=mask, pos_weights=None, neg_weights=None)
 
     @staticmethod
-    def collate(samples: Iterable[SLCWASampleType]) -> SLCWABatch:
+    def collate(samples: Iterable[SLCWABatch]) -> SLCWABatch:
         """Collate samples."""
+        samples = list(samples)
         # each shape: (1, 3), (1, k, 3), (1, k, 3)?
-        masks: LongTensor | None
-        positives, negatives, masks = zip(*samples, strict=False)
-        positives = torch.cat(positives, dim=0)
-        negatives = torch.cat(negatives, dim=0)
-        mask_batch: BoolTensor | None
-        if masks[0] is None:
-            assert all(m is None for m in masks)
-            mask_batch = None
-        else:
-            mask_batch = torch.cat(masks, dim=0)
-        return SLCWABatch(positives, negatives, mask_batch)
+        positives = torch.cat([s.positives for s in samples], dim=0)
+        negatives = torch.cat([s.negatives for s in samples], dim=0)
+        masks = _cat_or_none(s.masks for s in samples)
+        pos_weights = _cat_or_none(s.pos_weights for s in samples)
+        neg_weights = _cat_or_none(s.pos_weights for s in samples)
```

Comment:
typo check

Reply:
Yes!

Reply:
fixed in https://github.com/pykeen/pykeen/pull/1532/commits/4150009405676056f431b79c2f5ed74224127605